src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -405,5 +405,15 @@ public void Dispose()
             State = FunctionInvocationDispatcherState.Disposing;
             Dispose(true);
         }
+
+        public async Task RestartAsync()
+        {
+            // Dispose and restart all initialized channels
+            var channels = await GetInitializedWorkerChannelsAsync();
```

Comment:
I think you should consider adding
```
State = FunctionInvocationDispatcherState.WorkerProcessRestarting;
```

Since we're re-starting all of these processes, we don't want any new requests landing on these old workers. There is still a chance that a worker process restarts and sets state back to "Initialized" (line 125) before all workers are disposed (say we have 10), but this would be a start in preventing that.

Reply:
+1 For adding new state:WorkerProcessRestarting